src/freertos_drivers/net_cc32xx/CC32xxWiFi.hxx

```diff
@@ -208,6 +209,13 @@ public:
      * isthe function to execute.*/
     void run_on_network_thread(std::function<void()> callback);
 
+    /** Add and HTTP get token callback.
+     * @param callback the std::pair<> of the function to execute and the
+     *        matching token to execute the callback on
```

Comment:
could you describe what a "token" is? Also the documentation of the callback function needs to come here, i.e. when will the callback be executed and what should it return.
please document the lifetime requirements on the token const char*. How long does the memory pointed to stay alive? Who owns the memory?